elastic/kibana
# [Infrastructure UI] Use ad-hoc dataview in the Hosts View

## Summary

closes: [#152024](https://github.com/elastic/kibana/issues/152024)

This PR changes the Hosts View to use an ad-hoc Data View. 


### How to test 
- Go to `Infrastructure` > `Hosts`
- Use unified search
  - Save a query
  - Load a saved query
  - Clear all search criteria
  - Share a URL containing filters
  - Copy a URL with filters and open it in another tab
    - Edit a filter
- Use control filters
- Open a chart in Lens
- Check the data-view id
  - Apply a filter and check the `index` attribute in the `meta` object.
    - the id should look like `infra_metrics_UUID`  and should be the same each time the page is refreshed.


### For maintainers

Since ad-hoc data views are not saved, it needs to be created with a unique id to avoid possible naming collisions, at the same time, we don't want to create a new unique identifier each time an ad-hoc data view is created. A solution for this was to create a uuid id based on the index patterns configured in the settings.  

I've reviewed the way the Data View context was being used and removed a Data View object from being passed as props in favor of the context state. 

The default filters in Lens assets were also removed, as they were not necessary for the visualizations
